amethyst_ui/src/button/mod.rs

```diff
@@ -2,35 +2,66 @@ mod actions;
 mod builder;
 mod retrigger;
 mod system;
+mod text_color;
 
 pub use self::{
     actions::{UiButtonAction, UiButtonActionType},
     builder::{UiButtonBuilder, UiButtonBuilderResources},
     retrigger::{UiButtonActionRetrigger, UiButtonActionRetriggerSystem},
     system::UiButtonSystem,
+    text_color::UiButtonTextColor,
 };
-///! A clickable button.
-use amethyst_core::specs::prelude::{Component, DenseVecStorage};
 
-use serde::{Deserialize, Serialize};
+use crate::{
+    UiTransform, UiText,
+    Widget, widget_components, UiSoundRetrigger
+};
+use amethyst_core::{
+    Parent,
+    specs::prelude::Entity,
+};
+use amethyst_renderer::TextureHandle;
 
 /// A clickable button, this must be paired with a `TextureHandle`
 /// and this entity must have a child entity with a `UiText`.
-#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+#[derive(Debug, Clone)]
 pub struct UiButton {
-    /// Default text color
-    text_color: [f32; 4],
+    name: String,
```

Comment:
If the `Widgets` resource is a HashMap, you don't need to have the key here (name). Also, not having the key here allows to not require generics on UiButton.

UiButton<K> { id: K, image_entity.....}